### ISSUE TITLE
Maya: Reference Alembic produces inconsistencies with dag paths / namespaces

### ISSUE THREAD

**Issue by BigRoy** [pypeclub/OpenPype#2527]
### Issue

**Describe the bug**

_Sorry for a lengthy description, but let me take you through what happens._ :)

Loading Alembics into Maya through referencing fails to include the namespaces that are inside the Alembic file.

At the core this is more of an Alembic bug. It is basically [this issue](https://github.com/vvdoogy/alembic/issues/256) which should've been patched a long time ago and does work with `AbcImport` (just importing some Alembic geometry) but not when referencing.

Say I export an object to Alembic that has a namespace:
```
hero:cube_GEO
```

Now I reference this in Maya with Alembic. I reference it into a namespace, let's say it's a cache for a shot 010 and the namespace uses that. What I'd expect:
```
shot010:hero:cube_GEO
```
Instead I get:
```
shot010:cube_GEO
```
_Note: The actual .abc file DOES contain the namespaces + using `AbcImport` to import the content does add the namespace_

We don't rely all too much on namespaces and so far the actual object name seems alright. The issue isn't that big. Right? However, the issue becomes more complicated if I for example had the `cube_GEO` multiple times in a group with different namespaces. That's something that by default can happen quite easily with OpenPype, as loaded references get added into a root group called `{namespace}:_GRP` (thus `_GRP` will always conflict).

So say my source scene is like this. Similar to what OpenPype does during load (making `_GRP` top groups):

```
group/
    hero_01_:_GRP
    hero_02_:_GRP
    hero_03_:_GRP
    hero_04_:_GRP
```

I reference this output Alembic file. Maya gives me this:
```
shot010:group/
    shot010:_GRP
    shot010:_GRP1
    shot010:_GRP2
    shot010:_GRP3
```

Note the numbered suffices that Maya is giving it DURING the referencing in the MPxFileTranslator (see [Additional Context](#additionalcontext) below) due to conflicts in the loaded names.
Again, this problem might feel small.

But say the scene gets updated and another cube is added, maybe even from another asset.

Source:
```
group/
    hero_01_:_GRP
    hero_02_:_GRP
    villain_01_:_GRP
    hero_03_:_GRP
    hero_04_:_GRP
```

Result on reference:
```
shot010:group/
    shot010:_GRP
    shot010:_GRP1
    shot010:_GRP2
    shot010:_GRP3
    shot010:_GRP4
```

Now if I had reference edits on the `_GRP2` hierarchy then suddenly now they might be mismatched. We don't know exactly where "villain" ended up and names might have shuffled around in the loaded scene.

This could definitely introduce problems, like losing edits in a Lighting scene, etc.

**To Reproduce**

1. Go to OpenPype > Load..
2. Load a couple of pointcaches - feel free to mix up from different assets/subsets
3. You should have some namespaces with `{namespace}:_GRP` roots. Add these roots to a single group.
4. Go to OpenPype > Create..
5. Make a Pointcache of that top group.
6. Load the new pointcache.

The issue should be visible now.

Or a simple case that shows the issue through code:
```python
from maya import cmds
import openpype.hosts.maya.api.lib as lib
import tempfile


# Create some nodes
nodes = []
nodes.append(cmds.createNode("transform", name="namespace_01:_GRP"))
nodes.append(cmds.createNode("transform", name="namespace_02:_GRP"))
nodes.append(cmds.createNode("transform", name="namespace_03:_GRP"))
nodes.append(cmds.createNode("transform", name="namespace_04:_GRP"))

# Group together
group = cmds.group(name="group", empty=True)
cmds.parent(nodes, group)

with tempfile.TemporaryDirectory() as folder:
    path = os.path.join(folder, "test.abc")
    
    # Alembic Export
    cmds.select(group, noExpand=True, hierarchy=True)
    lib.extract_alembic(path, start=1, end=1, selection=True)

    # Reference Alembic
    nodes = cmds.file(path,
                      namespace="import",
                      sharedReferenceFile=False,
                      reference=True,
                      type="Alembic",
                      returnNewNodes=True)
```

### Solutions

It's a tricky thing because some of it is due to a potential bug in the Alembic plug-in and how it references. Since the reference loader seems to just be unable to do this correctly out of the box maybe we should ignore this.

We could also do it the other way around and Export with `-stripNamespaces` on extraction. However, that would mean the example scenes as shown above CANNOT be extracted like that. Either the `_GRP` names need to be changed OR they must be each put in another sub group to avoid conflict. 

We actually had that in our previous including [validating clashing node names](https://github.com/Colorbleed/colorbleed-config/blob/acre/colorbleed/plugins/maya/publish/validate_clashing_node_names.py) when `-stripNamespaces` was enabled. Our [Creators had `stripNamespaces` enabled by default](https://github.com/Colorbleed/colorbleed-config/blob/acre/colorbleed/plugins/maya/create/colorbleed_pointcache.py#L28-L29). In practice this meant the problem would be found sooner and resolved before becoming a problem down the pipeline - but still a hurdle for mostly layout/animation crew.

We also had conflicts much less due to our loaded content not all being in a `{namespace}:_GRP` but the part after the namespace also having e.g. subset or asset name. This meant that conflicts would really only occur when the same objects were in a group multiple times and stripNamespaces was enabled. 

### Additional context

Maya `AbcImport` plug-in registers a `MPxFileTranslator` so Maya can now reference that new file type. An file translator when used for referencing basically means that whatever is created during that `read` method is part of that single reference and gets the namespace appened.

The main logic for the [MPxFileTranslator and how it reads the content is here](https://github.com/alembic/alembic/blob/6253269ad35db108ed7ab26a70fb3c4183cb710e/maya/AbcImport/AlembicImportFileTranslator.cpp#L49-L51)

### Discussion

Any thoughts on this?

**Comment by antirotor** [comment]
This wasn't fixed by #2532 right?

**Comment by BigRoy** [comment]
No, this is still a problem with the Alembic importer in Maya it seems.

I did a quick test run in Maya 2023.3:

![afbeelding](https://user-images.githubusercontent.com/2439881/228961667-f8f5223a-d665-46bd-be54-edb2c5ee6d8a.png)

I exported the top cube with double namespaces as an Alembic with `-stripNamespaces` flag **disabled** (not set). As such the `.abc` file does contain the namespaces as intended. However, importing[^1] or referencing into Maya does NOT generate the namespaces.

[^1]: On import it actually _does_ generate the namespaces but ONLY if the namespaces already exist in the scene. This is also [described here](https://github.com/vvdoogy/alembic/issues/352#issue-123616032).
